iotdb-core/datanode/src/main/java/org/apache/iotdb/db/storageengine/dataregion/DataRegion.java

```diff
@@ -1240,7 +1254,14 @@ private void insertToTsFileProcessors(
 
     if (CommonDescriptor.getInstance().getConfig().isLastCacheEnable()) {
       if ((config.getDataRegionConsensusProtocolClass().equals(ConsensusFactory.IOT_CONSENSUS)
-          && insertRowsNode.isSyncFromLeaderWhenUsingIoTConsensus())) {
+              && insertRowsNode.isSyncFromLeaderWhenUsingIoTConsensus())
```

Comment:
I'm just wondering whether the `isSyncFromLeaderWhenUsingIoTConsensus` method can be replaced by `isGeneratedByRemoteConsensusLeader`.

Reply:
I think so, @OneSizeFitsQuorum may give some point?

Reply:
Besides, we can also consider the RatisConsensus here.

Reply:
yes we can do that now

Reply:
@OneSizeFitsQuorum Since `isSyncFromLeaderWhenUsingIoTConsensus` method can be replaced by `isGeneratedByRemoteConsensusLeader`, perhaps we can reduce the judgment conditions to only retain `isGeneratedByRemoteConsensusLeader`, and then maintain this variable in the consensus protocol of all DataRegions, which may be more elegant and concise.